### ISSUE TITLE
This software seems to be designed to spread malware

### ISSUE THREAD

**Issue by MiguelGuthridge** [V4NSH4J/discord-mass-DM-GO#485]
This software can easily used as a simple way to DM as many users as possible to encourage them to download malware, creating random burner accounts to do so. 

By the looks of things that's literally the purpose of this software, which begs the question, why would anyone who isn't a bad actor choose to facilitate such illegal activity?

Even when it's not used for the purpose, it is at the very least nuisanceware - I'm certain that the thousands of people who receive random unsolicited DMs from people using this software don't want to be spanned about discord server invites, or Telegram groups discussing the usage of this particular software, or other annoyances. Once again, this software is literally designed to be used for this purpose, so why would anyone who isn't a bad actor choose to facilitate such awfulness?

Basically, my issue is that this software shouldn't exist. I cannot think of a single ethical purpose that it can be used for.

**Comment by misty00x** [comment]
I'm sure it's used that way, but that doesn't mean everyone uses it for this purpose. By your logic we should ban the use of cell phones and computers because scammers/hackers use them for illegal activity.

I personally use this as a way to advertise my server to anyone that's interested in joining. Most of the servers I am in or projects I follow I found because some random bot DM'd me and I found interest in the project. If you don't want to get random DMs then disable DM's from non friends in your discord settings.

**Comment by MiguelGuthridge** [comment]
Unlike this software, computers and phones aren't designed for spamming people and giving them malware.

Your use case seems like spam to me. Do those people want to be randomly contacted by your account? I know I certainly didn't. I keep my DMs open because people often contact me for legitimate purposes - I shouldn't need to disable them to avoid getting random rubbish clogging up my inbox.

I frequently receive SMS messages from random phone numbers advertising random websites that I have no interest in. What makes your strategy of DMing thousands of people to advertise your software any different to their advertising (which is actually illegal due to anti-spam laws in my country)?

If you want to share your projects, post about them on Reddit (I've done this for a few of mine), or pay for ads through Google or Facebook - you'll be able to target specific demographics to make sure you're not annoying random people, and you won't frustrate so many people.

**Comment by MiguelGuthridge** [comment]
Literal screenshots from your Readme include rules for Discord servers that include preventing spam
![image](https://user-images.githubusercontent.com/31365175/161910360-3cb906d1-5483-4f49-a9f2-c7e0608b3561.png)

**Comment by V4NSH4J** [comment]
Hey @MiguelGuthridge thank you for the issue. 
This software DMDGO, was written back in November as a [proof of concept](https://github.com/V4NSH4J/discord-mass-DM-GO#q-will-you-mass-dm-for-me) on how bad actors launch large unsolicited DM advertisement campaigns. 
The DMDGO Authors do not encourage anyone to use this program for malicious purposes. You can read the full [disclaimer](https://github.com/V4NSH4J/discord-mass-DM-GO#disclaimer) here. 
We unfortunately can't track what the users do with the software and we refuse responsibility for any of it as we have no way of knowing and controlling. 
I'm sorry you have been receiving unsolicited Direct messages from Discord Userbots, but those could be from any software, this is just one of many that perform this functionality. Please make sure to:
1) Close your Discord DMs for that particular server where you're receiving spam
2) Reporting the account to Discord
3) Reporting the account to the server administration so that appropriate action can be taken.
4) You can close a spam DM by clicking the "X" in front of the name. 

I'm sorry to know that this may be illegal in your country, but DMDGO was written in India where there aren't any such laws. It was written in accordance with the law. Spam legislation is non-existent, the Information Technology act of 2000 does not cover anything related to it. I would strongly recommend against using this software if it's prohibited in your country. 

This is a small piece of a very big modern problem of bots taking over several social media platforms with false engagement and statistics. We hope to raise some awareness about that as well. 

Rest assured, the proof of concept was released so it could be patched by Discord, but as you can see, Discord didn't worry about it for 6 months. You can try writing them an email regarding this. 

I don't know what you mean when you say that half the features are unethical. The project is in accordance with GitHub's rules as well. Again, most of it's functions are a violation of Discord Terms of Service and Community Guidelines and we neither encourage you nor recommend anyone to use the software for spamming. We intended Discord to patch it, 

For your last image, I unfortunately can't do anything if a user decides to use this software to spam a server where it's not allowed. There's no way for me to track and stop it. When we used to function on Discord, we made sure to report everyone in violation of Discord Terms of Service and Community Guidelines and encouraged people to adhere by it and you should too.

If you have any other questions feel free to drop them.

**Comment by MiguelGuthridge** [comment]
Thanks for the clarification. It makes sense for it to be a proof-of-concept. Perhaps it could be a good idea to be clear about that in the readme. Discord definitely has to do something about its spam problem, so hopefully projects like this should help push them to act more. I think the main issue is that the fact that this is a proof-of-concept isn't clear when reading the Readme then.

**Comment by V4NSH4J** [comment]
> Thanks for the clarification. It makes sense for it to be a proof-of-concept. Perhaps it could be a good idea to be clear about that in the readme. Discord definitely has to do something about its spam problem, so hopefully projects like this should help push them to act more. I think the main issue is that the fact that this is a proof-of-concept isn't clear when reading the Readme then.

Yeah, there used to be a whole paragraph dedicated to it, I rewrote the readme sometime in between and forgot to include it again. I apologize for that. Rest assured, Discord has only started acting seriously on it's spam problem since this week, let's hope to see some positive change.
